guineaPigFacts/src/test/java/com/factsdemo/guineaPigFacts/controllerTests/UserControllerTest.java

```diff
@@ -0,0 +1,225 @@
+import com.factsdemo.guineaPigFacts.services.UserService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(UserController.class)
+public class UserControllerTest {
+    @Autowired
+    private MockMvc mockMvc;
+    @MockBean
+    private UserService userService;
+
+    private final String goodUserId = "100000000000000000000001";
+    private final String badUserId = "1";
+    private final String goodUserName = "testUser_1";
+    private final String badUserName = "t11#";
+    private final String goodUserPassword = "testPassword1";
+    private final String badUserPassword = "testPassword";
+
+    private final String goodUserEmail = "test@gmail.com";
+    private final boolean dailyUpdate = true;
+
+    private Contact goodContact;
+    private User goodUser;
+    private User badUser;
+
+    private String errorMessageId = "";
+    List<User> userArray = new ArrayList<User>();
+
+    @Before
+    public void setup() {
+        errorMessageId = "User "+ badUserId +  " was not found";
+        goodContact = new Contact(goodUserEmail, dailyUpdate);
+        goodUser = new User(goodUserName, goodUserPassword, goodContact);
+        goodUser.setId(goodUserId);
+        badUser = new User(badUserName, badUserPassword, goodContact);
+    }
+
+    /**
+     * Get controller test '/user/{id}' with a userId that is mocked to exist
+     */
+    @Test
+    public void getByIdControllerTest_ShouldReturnStatusOk() throws Exception {
+        when(userService.findById(goodUserId)).thenReturn(java.util.Optional.ofNullable(goodUser));
+        mockMvc.perform(get("/user/{id}", goodUserId)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.userName").value(goodUser.getUserName()));
+    }
```

Comment:
This test is fine, but I just wanted to point out that best practice is to test one thing per test. This method tests the content type, status, and response body.  I don't think it's necessary to change this, but it's something to keep in mind if you have more complicated logic.  For example, if you had some logic that transformed the response, you might want to split that out into a separate unit test.  

Keeping your unit tests well scoped to a single assertion allows you to more quickly find where your code is broken. (ie - one test fails and you know exactly what it's testing rather than several somewhat related tests failing)